File: app/src/main/java/com/abishov/hexocat/commons/network/HexocatAdapterFactory.java
package com.abishov.hexocat.commons.network;

import android.support.annotation.NonNull;

import com.google.gson.TypeAdapterFactory;
import com.ryanharter.auto.value.gson.GsonTypeAdapterFactory;

@GsonTypeAdapterFactory
public abstract class HexocatAdapterFactory implements TypeAdapterFactory {

    // Static factory method to access the package
    // private generated implementation
    @NonNull
    public static TypeAdapterFactory create() {
        return new AutoValueGson_HexocatAdapterFactory();
    }
}

File: app/src/main/java/com/abishov/hexocat/common/network/HexocatAdapterFactory.kt
package com.abishov.hexocat.common.network

import com.google.gson.TypeAdapterFactory
import com.ryanharter.auto.value.gson.GsonTypeAdapterFactory

@GsonTypeAdapterFactory
abstract class HexocatAdapterFactory : TypeAdapterFactory {

  companion object {
    fun create(): TypeAdapterFactory {
      return AutoValueGson_HexocatAdapterFactory()
    }
  }
}
